constructor/osx/update_path.sh

```diff
@@ -8,4 +8,35 @@ set -eux
 
 PREFIX="$2/__NAME_LOWER__"
 PREFIX=$(cd "$PREFIX"; pwd)
-"$PREFIX/bin/python" -m conda init --all
+INIT_FILES=$("$PREFIX/bin/python" -m conda init --all | tee)
+
+# Just like in run_install.sh, the files generated by the installer
+# are owned by root when installed outside of $HOME. So, ownership of
+# files modified by conda init must be changed to belong to $USER.
+# For shells like fish or powershell, conda init creates subdirectories
+# inside $HOME and/or $PREFIX, so ensure that parent directories also
+# have the correct owner.
+if [[ "${USER}" != "root" ]]; then
+    echo "Fixing permissions..."
+    read -r -a MODIFIED_FILES <<< "$(\
+      echo "${INIT_FILES}" |\
+      grep modified |\
+      awk '{print $2}' |\
+      # Only grab files inside $HOME or $PREFIX.
+      # All init files should be there, but that may change, and it
+      # is better to miss files than to have an infinite loop below.
+      grep -E "^(${HOME}|${PREFIX})"\
+    )"
+    for file in "${MODIFIED_FILES[@]}"; do
+        while [[ "${file}" != "${HOME}" ]] && [[ "${file}" != "${PREFIX}" ]]; do
+            # Check just in case the file wasn't created due to flaky conda init
+            if [[ -f "${file}" ]] || [[ -d "${file}" ]]; then
+                OWNER=$(stat -f "%u" "${file}" | id -un)
+                if [[ "${OWNER}" == "root" ]]; then
+                    chown "${USER}" "${file}"
+                fi
+            fi
+            file="${file%/*}"
+        done
+    done
+fi
```

Comment:
Awesome shell skills here! This script is called with `sh` though, so I think we need to either:

- Change the shebang to bash
- Make sure we only use strict POSIX (e.g. no `[[ ]]` ifs).

Reply:
Doesn't `sh` link to `bash` on MacOS?

Reply:
Even if it does, I think it's better to be explicit. I changed the shell to bash since that's what we're using in the other scripts as well.